### ISSUE TITLE
Ghost rows are added to the table when sorting, editing, (etc?)

### ISSUE THREAD

**Issue by uap-universe** [dj-fiorex/angular2-smart-table#55]
https://github.com/dj-fiorex/angular2-smart-table/blob/4270df984f6b27cb35d5f4131ec5aa315a04464d/projects/angular2-smart-table/src/lib/lib/data-set/data-set.ts#L34

@hexdecimal16 @dj-fiorex 

This push operation can add ghost rows to the table, e.g. when sorting or inline editing. Maybe there are more operations that lead to this incorrect behavior. We should fix this asap, because this is a critical bug.

Please, in the future, do not merge (and more importantly do not release) untested changes that quickly. I know this whole npm community has a release first, fix second mentality, but since this table may be used in production code by many people, we should be more careful.

**Comment by hexdecimal16** [comment]
see https://github.com/dj-fiorex/angular2-smart-table/commit/4270df984f6b27cb35d5f4131ec5aa315a04464d#commitcomment-72858705

**Comment by hexdecimal16** [comment]
> @hexdecimal16 are you already resolved the bug?

I think so, let @uap-universe confirm

**Comment by uap-universe** [comment]
@hexdecimal16 I looked deeper into the changes from the recent days and discovered this commit: https://github.com/dj-fiorex/angular2-smart-table/commit/3a5972fba474883f7853fc81bfe5502a8eae7267. The commit message describes the behavior we are observing and suggests that you implemented that on purpose as a "feature". Maybe you can elobarote a bit on this, so I can understand what your intention was. From my perspective right now this is a breaking change.

**Comment by hexdecimal16** [comment]
I am not able to reproduce it. I am attaching screenshots for reference
Initial state:
![Screenshot from 2022-05-04 20-17-08](https://user-images.githubusercontent.com/47829719/166707228-f35e0891-efd0-499c-9939-8feb32bde564.png)
Selecting one row:
![Screenshot from 2022-05-04 20-18-44](https://user-images.githubusercontent.com/47829719/166707388-48fd3f91-cdd4-48a4-b9ac-0cd57d1961d6.png)
Sorting ascending:
![Screenshot from 2022-05-04 20-19-22](https://user-images.githubusercontent.com/47829719/166707560-0faccbde-db4e-4fcb-803d-fba4372640b3.png)
Sorting descending:
![Screenshot from 2022-05-04 20-19-56](https://user-images.githubusercontent.com/47829719/166707675-4525c053-30e3-44a8-8948-fdf6bd9cac22.png)

**Comment by uap-universe** [comment]
@hexdecimal16 I played around with the demo application a bit and I cannot reproduce the bug there. But I also observed that in the demo application the selected row (and by selected I mean the row that is highlighted in grey, *not* necessarily a row that is ticked via the checkbox!) changes when sorting. This is not the case in may live application. There the selected (i.e. grey highlighted row) does not change, but instead moves to the top of the table. I'll try to analyze...

Before:
![image](https://user-images.githubusercontent.com/733505/166711319-800c53cc-9994-4732-9da7-4d60be30acd6.png)

After:
![image](https://user-images.githubusercontent.com/733505/166711396-d3cfdef5-ca55-4a38-8dbe-757b7c89be7f.png)
